lib/shanty_template_plugin/template_plugin.rb

```diff
@@ -0,0 +1,85 @@
+require 'deep_merge'
+require 'erubis'
+require 'fileutils'
+require 'shanty/plugin'
+require 'yaml'
+
+module ShantyTemplatePlugin
+  # Public: Shanty template plugin
+  class TemplatePlugin < Shanty::Plugin
+    PROJECT_EXTENSION = 'erb'
+    DS_EXTENSTION = 'syaml'
+
+    option :datasource, 'plain'
+    option :outputdir, 'build'
+    subscribe :build, :on_build
+    projects "**/*.#{PROJECT_EXTENSION}"
+
+    def on_build(project)
+      load_data(project).each do |env, data|
+        project_files(project).each do |template|
+          output = output_filename(project, template, env)
+          FileUtils.mkdir_p(File.dirname(output))
+          File.open(output, 'w') do |out|
+            out.write(Erubis::Eruby.new(File.read(template)).result(data: data))
```

Comment:
Could just use `File.write` instead for this, it'll be faster because it doesn't need to execute a block or wrap the file handle in a Ruby object:

```ruby
File.write(output , Erubis::Eruby.new(File.read(template)).result(data: data))
```